### PR TITLE
Add TrainingDataWithPremises Extraction

### PR DESCRIPTION
**Pull request by JOSHCLUNE**

### CODE REVIEW

File: scripts/run_pipeline.py [hunk 1]
```diff
@@ -33,32 +35,44 @@ def _run_cmd(cmd, cwd, input_file, output_file):
...
-        cmd='training_data',
-        cwd=cwd,
-        input_file=input_module,
-        output_file=os.path.join(
-            output_base_dir, 
-            DIR_NAMES['training_data'],
-            _get_stem(input_module, input_file_mode) + '.jsonl'
-        )
-    )
+    # _run_cmd(
```
Comment (wellecks) [inline scripts/run_pipeline.py]: Perhaps we can add a command line flag to determine which of these is run (rather than commenting out)

**Review by wellecks — CHANGES_REQUESTED** [review]
Left a comment about a command-line option for `run_pipeline.py`. Otherwise this looks great, thanks!

**Comment by wellecks** [comment]
Also, could you add a short description of the new commands to the readme?

**Comment by JOSHCLUNE** [comment]
I've added command-line options for run_pipeline and updated the README to explain them in the most recent commit. Further additions (such as adding a field for all premises used for the current declId proof or a field for all premises from the current state to the end of the goal) will be handled in a later PR once I am able to address them. Let me know if there are any additional issues, thanks.

**Comment by wellecks** [comment]
Awesome, thanks!

**Pull request merged**